bootloader/bootloader.go

```diff
@@ -310,3 +311,132 @@ func removeKernelAssetsFromBootDir(bootDir string, s snap.PlaceInfo) error {
 
 	return nil
 }
+
+// kernelCommandLineSplit tries to split the string into a list of elements that
+// would be passed by the bootloader as the kernel command arguments.
+//
+// See https://www.kernel.org/doc/html/latest/admin-guide/kernel-parameters.html for details.
+func kernelCommandLineSplit(s string) (out []string, err error) {
+	const (
+		argNone            int = iota // initial state
+		argName                       // looking at argument name
+		argAssign                     // looking at =
+		argValue                      // looking at unquoted value
+		argValueQuoteStart            // looking at start of quoted value
+		argValueQuoted                // looking at quoted value
+		argValueQuoteEnd              // looking at end of quoted value
+	)
+	var b bytes.Buffer
+	var rs = []rune(s)
+	var last = len(rs) - 1
+	var errUnexpectedQuote = fmt.Errorf("unexpected quoting")
+	var errUnbalancedQUote = fmt.Errorf("unbalanced quoting")
+	var errUnexpectedArgument = fmt.Errorf("unexpected argument")
+	var errUnexpectedAssignment = fmt.Errorf("unexpected assignment")
+	// arguments are:
+	// - arg
+	// - arg=value, where value can be any string, spaces are preserve when quoting ".."
+	var state = argNone
+	for idx, r := range rs {
+		maybeSplit := false
+		switch state {
+		case argNone:
+			switch r {
+			case '"':
+				return nil, errUnexpectedQuote
+			case ' ':
+				maybeSplit = true
+			default:
+				state = argName
+				b.WriteRune(r)
```

Comment:
Here we have the case of a parameter like `==a` which I think should be invalid (but I didn't see if it actually is, kernel-wise).

Reply:
Also a lone `=` could pass, but I don't know what the kernel thinks about that.